feat(data): Making use of the new scanner info types

Zitchas opened the pull request:
Adds the:
- Fuel scanner (fuel)
- Patrol Scanner (crew and weapon range)
- Rangefinder Scanner (Rangefinder)
- Thermal Scanner (thermal)

Modifies:
- Tactical scanner (velocity and energy)
- Surveillance pod (cargo+outfit scan, crew, energy, and fuel)
- All the scanning ranges have been buffed slightly. Probably an insignificant amount, but just a bit, anyway.

Outfitters:
- Thermal is added to outfiters at the same time as the flamethower.
- Rangefinder is added to common. Others are added to a variety of advanced, security, and pirate outfitters.

To do:
- [ ] Hai scanners
- [x] Remnant scanners
- [x] Coalition scanners
- [x] Any other scanners (Couldn't find other passive scanner outfits, just a few built-in things I didn't want to touch)

TheGiraffe3 commented on data/human/outfits.txt:
```diff
@@ -232,30 +232,70 @@ outfit "Asteroid Scanner"
+	"mass" 2
+	"outfit space" -2
+	"fuel scan power" 35
+	description "Knowing how much fuel your target has available is useless to most ship captains, but can be very valuable to both law enforcement and pirates. Sometimes the ocassional merchant uses one to determine if they are likely to be able to outlast someone in a long pursuit."
```
```suggestion
	description "Knowing how much fuel your target has available is useless to most ship captains, but can be very valuable to both law enforcement and pirates. Sometimes the occasional merchant uses one to determine if they are likely to be able to outlast someone in a long pursuit."
```

TheGiraffe3 commented on data/human/outfits.txt:
```diff
@@ -232,30 +232,70 @@ outfit "Asteroid Scanner"
+	"outfit space" -2
+	"crew scan power" 35
+	"weapon scan power" 35
+	description "Patrol ships, law enforcement, and immigration officials in the current stellar environemnt are generally concerned with two things: How many people are on board, and how far away does one need to be in order to stay safe. As such, these scanners are primarily found on ships who are responsible for tracking the flow of people and ocassionally conducting arrests or inspections."
```
```suggestion
	description "Patrol ships, law enforcement, and immigration officials in the current stellar environment are generally concerned with two things: how many people are on board, and how far away does one need to be in order to stay safe. As such, these scanners are primarily found on ships who are responsible for tracking the flow of people and occasionally conducting arrests or inspections."
```

TheGiraffe3 commented on data/human/outfits.txt:
```diff
@@ -232,30 +232,70 @@ outfit "Asteroid Scanner"
+	"mass" 1
+	"outfit space" -1
+	"range finder power" 1
+	description "While named a scanner for simplicity, this is more of an analysis and display module that utilizes standard system detection equipment to determine exact ranges to target ships. Once installed, the distance to the currently targeted ship will be display in the panel next to it."
```
```suggestion
	description "While named a scanner for simplicity, this is more of an analysis and display module that utilizes standard system detection equipment to determine exact ranges to target ships. Once installed, the distance to the currently targeted ship will be displayed in the panel next to it."
```

TheGiraffe3 commented on data/human/outfits.txt:
```diff
@@ -232,30 +232,70 @@ outfit "Asteroid Scanner"
+	"mass" 2
+	"outfit space" -2
+	"thermal scan power" 35
+	description "While thermal signatures have always been useful as a means of identifying and tracking ships, determining a ship's effective average thermal load was functionally useless until recently. With the development of the plasma canon and then the flamethrower, attempting to overheat a ship has become a viable combat technique. As such, engineers with the Free World have begun assembling  ...
```
```suggestion
	description "While thermal signatures have always been useful as a means of identifying and tracking ships, determining a ship's effective average thermal load was functionally useless until recently. With the development of the plasma cannon and the flamethrower, attempting to overheat a ship has become a viable combat technique. As such, engineers with the Free World have begun assembling these sensor packages to support the daring captains who choose to use this tactic."
```

Zitchas replied:
I included the "then" in this case because the development of the plasma cannon and flamethrower were sequential and different things, not part of the same work. The former has been around for significantly longer than the flamethrower, while still being a recent invention, and I think having the "then" conveys this better.

TheGiraffe3 replied:
True fact.
```suggestion
	description "While thermal signatures have always been useful as a means of identifying and tracking ships, determining a ship's effective average thermal load was functionally useless until recently. With the development of the plasma cannon and then the flamethrower, attempting to overheat a ship has become a viable combat technique. As such, engineers with the Free World have begun assembling these sensor packages to support the daring captains who choose to use this tactic."
```

Zitchas commented:
Thanks for reviewing the text. It is much appreciated.

TheGiraffe3 commented:
Always happy to help.

TheGiraffe3 commented on data/remnant/remnant outfits.txt:
```diff
@@ -632,14 +632,22 @@ outfit "Research Laboratory"
 	"unplunderable" 1
 	description "From the very beginning, the Remnant have relied on their ability to study everything they encounter. This knowledge has enabled them to survive and even prosper in the harsh environment of the Ember Waste. To enable this pursuit of knowledge, the Remnant designed this compact laboratory that is built onto the side of a cargo bay, mess hall, or other larger room. When the room isn't ...
 	description "	While planetbound scientists might consider it little more than a closet packed with scanners and equipment, to the Remnant it is a base of operations used to investigate alien and stellar phenomena on-site. For captains, having advanced equipment to support their sensor suites as well as a trained scientist to serve in their crew amount to a strong benefit."
+	description "	This laboratory requires a trained scientist to be permanently assigned to it, and as such increases your required crew by 1. It also continously consumes a small amount of energy."
```
```suggestion
	description "	This laboratory requires a trained scientist to be permanently assigned to it, and as such increases your required crew by 1. It also continuously consumes a small amount of energy."
```

Zitchas replied:
Thanks, I just got it myself.

Zitchas commented:
@Hurleveur @MasterOfGrey 

Given that the two of you are the primary people working on the Hai, are there any changes you would like to make to their scanner? Right now the value detector is quite an effective long-range cargo scanner and the same range as the tactical scanner for the generic "tactical" set of scans. 

My main thought is wondering if you want to focus on a few tactical elements instead of just having the default "tactical" set. 

For comparison, I'll note that the human "Tactical Scanner" has been shrunk down to 2 tons, but also only provides velocity and energy info. The other information elements have been moved off to new scanners, giving people the ability to avoid cluttering things with information they don't want and potentially get better range for the information they *do* want.

For reference, the options are:

- "acceleration scan power" : Current maximum potential acceleration
- "crew scan power" : Current crew
- "energy scan power" : Current energy
- "fuel scan power" : Current fuel
- "maneuver scan power" : Current maximum potential turn rate
- "range finder power" : distance from player's flagship to target
- "thermal scan power" : Current heat
- "velocity scan power" : Current speed
- "weapon scan power" : Displays the range of the longest-ranged gun and the longest-ranged turret

Given the description of the Value Detector, energy is an obvious fit; but I'm not sure exactly what others would fit the "more detailed tactical information". Are they focused on boarding strength? The ability of the target to run? 

I think I'd lean towards energy, crew, and weapons; although Hai weaponry outranges a lot of things, so I could see them not being as concerned with that and opting for, say, velocity and fuel.

So by my reading, I'd recommend the following, and bumping them up to about 60, with the energy one being higher, maybe 90? (I suggest these values because if the player wants to get just one of these (other than energy), they can get better range filling the same space with a human scanner, but if they want all of these or the energy in particular, then this Hai one is clearly superior)
- crew
- energy
- fuel
- velocity

Zitchas commented:
I'm noticing that most other groups don't have passive scanner outfits. One Sestor ship has built-in scanning, and there's a few other things here and there, but very little, really. I'm not going to add scanning capabilities to those factions that don't already have it.

MasterOfGrey commented:
> @Hurleveur @MasterOfGrey
> 
> Given that the two of you are the primary people working on the Hai, are there any changes you would like to make to their scanner?

Without reviewing the existing content in detail, some logical considerations:

UHai would want to know:
- Current Energy
- Cargo
- Distance

Given what they face off against these would cover their interactions with the Hai generally.
- Weapon Ranges would then make sense for vs wanderers - because they'd want to know if the birds started bringing something to upset them.


Hai Military would want to know:
- Acceleration
- Energy
- Fuel
- Distance
- Heat
- Weapon Ranges

These would be a holdover from the Korath raids, and would be sufficient to cover concerns re the UHai + humanity.


Hai Security would want to know:
- Acceleration
- Fuel
- Distance
- Crew
- Cargo

These would be key to allowing them to judge the nature of a vessel, whether they should be intercepting them, and whether they could successfully do so.



How you choose to apply these considerations I'll leave up to you.

TheGiraffe3 commented:
Looking at it, does the Research Laboratory really do anything now that most Remant ships have scanning built-in?

Hurleveur commented:
I agree with MasterOfGrey but I think the UHai would also like to know acceleration capability when they're hunting for cargo. That said they wouldnt need a very detailed version of it either

Fuel would only matter if its fully depleted and they can observe ships for that info (though for the speed as well tbf)

I dont know if crew would matter that much? Like yes to intimidate the others you'd want to be more numerous but thats often predictable with the ship sizes?

Zitchas commented:
Thanks for the info and points to consider!

> Looking at it, does the Research Laboratory really do anything now that most Remant ships have scanning built-in?

Scanning stacks, so, as it says in the Salvage Scanner description, while it isn't strictly speaking necessary in order to have that passive scanning capability, it does provide a significant boost to range. Most Remnant ships have a passive scanning range of somewhere in the 500-700 range, which is a shorter distance than many weapons. So if the captain wants to be able to see info about their target *before* they start getting shot at, the Laboratory is a necessity. 

In other words, it depends on how use data. If you just want to see the impacts of your weaponry (seeing how slow is impacting their turning and acceleration; or seeing how effective a thermal weapon is), then you are correct and the Laboratory isn't needed. If you want to be able to see some info about the ship *before* you engage so you have the option to think about your tactics before combat starts, then it's a necessity against most ships. 

> How you choose to apply these considerations I'll leave up to you.

All these are good points, and generally boil down to the fact that, like any good military, they'd really like to know everything. 

Regarding the weapon distance, my question would be "Does the value detector predate the UHai's conflict with the Wanderers?" If so, then this isn't really something that should affect it much, and might be grounds for a post-conflict-start development of a new sensor. 

Based on the above comments, my thoughts:

Unchanged:
- Cargo scanning
- Range finding
- Fuel

Buffed:
- Energy levels

Removed:
- Crew
- Thermal

Added:
- Acceleration

(Although when I say "stays the same" I mean "only buffed a little bit" in the case of fuel scanning, the change from 35 to 42 bumps the range from 591 up to 648. The energy info starts coming in at 974, and they can start scanning at the same distance as before: 866. Just a note: While rangefinding is listed with a power, it's just for consistency, as well as to minimize work if someone implements range limits on it in the future. Right now it's basically a binary, either 0 or anything positive. 0 is no rangrefinding, positive means they have it and it works to any range.)

This results in:
```
	"cargo scan power" 75
	"cargo scan efficiency" 10
	"acceleration scan power" 42
	"energy scan power" 95
	"fuel scan power" 42
	"range finder power" 42
```

Zitchas commented:
Alright, with Hurleveur's thumbsup on that, I'm going to merge this. That being said, if either of you (or, for that matter, authors/faction leads of other factions) have some insight into scanners in their area, I'm always open to doing some changes to make use of these. 

Also, a note on tech levels. A lot of this scanner information is kind of ambiguous as to how advanced a faction needs to be to have it. I'll use agility and acceleration as an example. I'd frankly expect that any spacefaring civilization should probably be able to analyze a ship's movement patterns and how it is changing velocity and position to determine what its thrust and turn speed are currently. This is basic levels that we can do to a certain degree in the present day here on Earth. On the other hand, these types of scanning *could* be analyzing the mechanics of the ship to determine what its capabilities are, which would be a very advanced ability. Most of the time, these are basically indistinguishable from each other. The only time it really appears is if one is observing a ship that is just coasting through space. The low-tech approach of observation and analysis shouldn't know anything, whereas the high-tech Star Trek style scanning info could. But how often do ships just coast anywhere? So the only real impact simple vs advanced tech has in this realm is in the lore and story of how it is portrayed. Probably also in the range of how far away it can be done, too.
